modules/programmingLanguages/go/default.nix

```diff
@@ -0,0 +1,18 @@
+{ mode, config, pkgs, lib, soxin, ... }:
+
+with lib;
+{
+  config.soxin.programmingLanguagesModules.go = (mkMerge [
+    {
+      vscode = {
+        extensions = [ pkgs.vscode-extensions.golang.Go ];
+      };
+    }
+
+    /*(optionalAttrs (mode == "home-manager") {
+      programs.go = {
+      enable = true;
+      };
+      })*/
```

Comment:
```suggestion
```

Let's open an issue for how we want to deal with modules related to programming languages, or let's discuss it in the RFC 3 PR.